CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Onboarding/OnboardingViewController.swift

```diff
@@ -103,23 +125,63 @@ class OnboardingViewController: UIViewController {
         dismiss(animated: true)
     }
 
-    private func nextPage() {
+    @objc private func pageControlSelectionAction(_ sender: UIPageControl) {
+        let index = IndexPath(item: sender.currentPage, section: 0)
+        pagesCollection.scrollToItem(at: index, at: .centeredHorizontally, animated: true)
+    }
+
+    @objc private func nextPage() {
+        if dataSource.currentPage < dataSource.itemSections.count - 1 {
+            let index = IndexPath(item: dataSource.currentPage + 1, section: 0)
+            pagesCollection.scrollToItem(at: index, at: .centeredHorizontally, animated: true)
+        } else {
+            close()
+        }
     }
 
     private func skip() {
         close()
     }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        if #available(iOS 13, *) {
+            view.layoutSubviews()
+        } else {
+            let offsetX = CGFloat(max(dataSource.currentPage, 0)) * (size.width)
+            coordinator.animate(
+                    alongsideTransition: { _ in self.pagesCollection.collectionViewLayout.invalidateLayout()
+                    },
+                    completion: { _ in
+                    self.pagesCollection.setContentOffset(CGPoint(x: offsetX, y: 0), animated: true)
+                        self.pagesCollection.reloadData()
+                }
+            )
+        }
+    }
+
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        pagesCollection.collectionViewLayout.invalidateLayout()
+    }
 }
 
 extension OnboardingViewController: OnboardingScreen {
     func didScroll(page: Int) {
+        if page == dataSource.itemSections.count - 1 {
+            navigationItem.rightBarButtonItem = nil
+            nextButton.setTitle("Get Started", for: .normal)
+        } else {
+            navigationItem.rightBarButtonItem = skipButton
+            nextButton.setTitle("Next", for: .normal)
```

Comment:
[Suggestion] Localization?

Reply:
it will be in a separate ticket - isn't it @chriskryn?

Reply:
@zladzeyka  I have seen we have PIA-3006 for this.